core/src/main/java/org/infinispan/protostream/descriptors/ProtoLock.java

```diff
@@ -0,0 +1,384 @@
+package org.infinispan.protostream.descriptors;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+/**
+ * Reads/writes <a href="https://github.com/nilslice/protolock">protolock</a> files.
+ */
+public class ProtoLock {
```

Comment:
There's a a few bits of duplicated code between the `writeMessage` and `writeEnum` being highlighted by Intellij, we should extract these to methods.

Reply:
Done